deployment/keystone/ocr3config.go

```diff
@@ -265,24 +313,72 @@ func configureOCR3contract(req configureOCR3Request) (*configureOCR3Response, er
 		return nil, fmt.Errorf("failed to generate OCR3 config: %w", err)
 	}
 	if req.dryRun {
-		return &configureOCR3Response{ocrConfig}, nil
+		return &configureOCR3Response{ocrConfig, nil}, nil
 	}
-	tx, err := req.contract.SetConfig(req.chain.DeployerKey,
-		ocrConfig.Signers,
-		ocrConfig.Transmitters,
-		ocrConfig.F,
-		ocrConfig.OnchainConfig,
-		ocrConfig.OffchainConfigVersion,
-		ocrConfig.OffchainConfig,
-	)
-	if err != nil {
-		err = DecodeErr(kocr3.OCR3CapabilityABI, err)
-		return nil, fmt.Errorf("failed to call SetConfig for OCR3 contract %s: %w", req.contract.Address().String(), err)
-	}
-	_, err = req.chain.Confirm(tx)
-	if err != nil {
-		err = DecodeErr(kocr3.OCR3CapabilityABI, err)
-		return nil, fmt.Errorf("failed to confirm SetConfig for OCR3 contract %s: %w", req.contract.Address().String(), err)
+	var proposal *timelock.MCMSWithTimelockProposal
+	if !req.useMCMS {
+
+		tx, err := req.contract.SetConfig(req.chain.DeployerKey,
+			ocrConfig.Signers,
+			ocrConfig.Transmitters,
+			ocrConfig.F,
+			ocrConfig.OnchainConfig,
+			ocrConfig.OffchainConfigVersion,
+			ocrConfig.OffchainConfig,
+		)
+		if err != nil {
+			err = DecodeErr(kocr3.OCR3CapabilityABI, err)
+			return nil, fmt.Errorf("failed to call SetConfig for OCR3 contract %s: %w", req.contract.Address().String(), err)
+		}
+		_, err = req.chain.Confirm(tx)
+		if err != nil {
+			err = DecodeErr(kocr3.OCR3CapabilityABI, err)
+			return nil, fmt.Errorf("failed to confirm SetConfig for OCR3 contract %s: %w", req.contract.Address().String(), err)
+		}
+	} else {
+		// use MCMS
+		configRequest, err := req.contract.SetConfig(
+			deployment.SimTransactOpts(),
+			ocrConfig.Signers,
+			ocrConfig.Transmitters,
+			ocrConfig.F,
+			ocrConfig.OnchainConfig,
+			ocrConfig.OffchainConfigVersion,
+			ocrConfig.OffchainConfig)
+		if err != nil {
+			err = DecodeErr(kocr3.OCR3CapabilityABI, err)
+			return nil, fmt.Errorf("failed to simulate SetConfig for OCR3 contract %s: %w", req.contract.Address().String(), err)
+		}
+		o := timelock.BatchChainOperation{
+			ChainIdentifier: mcms.ChainIdentifier(req.chain.Selector),
+			Batch: []mcms.Operation{
+				{
+					// Enable the source in on ramp
+					To:    req.contract.Address(),
+					Data:  configRequest.Data(),
+					Value: big.NewInt(0),
```

Comment:
what is value? no docs i could find. using same pattern as ccip `add_chain` [here](https://github.com/smartcontractkit/chainlink/blob/c1ee7ab715b524df6b580593e18f51637bd1500d/deployment/ccip/changeset/cs_add_chain.go#L39)

Reply:
its the native ETH transfer with the call (see call docs here https://www.evm.codes/?fork=cancun#f1), very very rarely non-zero